lib/ansible/plugins/inventory/__init__.py

```diff
@@ -0,0 +1,160 @@
+# (c) 2017,  Red Hat, inc
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import hashlib
+import string
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_bytes, to_text
+
+class BaseInventoryPlugin(object):
```

Comment:
I started that way, but since some of the methods do actually implement code I thought it was not appropriate anymore.